### ISSUE TITLE
Illegal instruction 4 with old Mac

### ISSUE THREAD

**Issue by ntustison** [ANTsX/ANTs#764]
_Just in case anybody else runs across this issue_--

I have an old Mac Pro (Early 2009).  Any binary from compiling the latest ANTs gives me the error "Illegal instruction: 4".  I traced it to [this change](https://github.com/ANTsX/ANTs/commit/dc50ca7dae7c3627b8931f589f70bd241fb612db).  Using the [previous version of the file](https://raw.githubusercontent.com/ANTsX/ANTs/616c58f36458ea125b4ceccf552396b350ded346/CMake/ITKSetStandardCompilerFlags.cmake) corrects this problem.  Based on the [commit message](https://github.com/ANTsX/ANTs/commit/dc50ca7dae7c3627b8931f589f70bd241fb612db), I've tried changing the optimization flags to 

```
  -DANTs_C_OPTIMIZATION_FLAGS:STRING="-march=native"
  -DANTs_CXX_OPTIMIZATION_FLAGS:STRING="-march=native"
  -DANTs_C_WARNING_FLAGS:STRING="-Wall -Werror"
  -DANTs_CXX_WARNING_FLAGS:STRING="-Wall -Werror"
```

and

```
  -DANTs_C_OPTIMIZATION_FLAGS:STRING="-march=native"
  -DANTs_CXX_OPTIMIZATION_FLAGS:STRING="-march=native"
```

but neither attempt caused things to work.

**Comment by cookpa** [comment]
This has come up a couple of times for users lately. I think it may indirectly depend on the OS because different compilers might make a different guess about the optimizer flags, but the underlying issue is wrong optimization flags being set for the CPU. 

I don't think the command line options will work because the error arises in ITK, and command line CMake options don't get passed through to the external projects in a Superbuild. 

There is apparently a way to fix this

  https://cmake.org/pipermail/cmake/2018-January/067002.html

**Comment by gdevenyi** [comment]
To be clear, the devs believe the issue is that your CPU is older than corei7, which is what our default "arch" setting is:
/CMake/ITKSetStandardCompilerFlags.cmake@master#L245 

You can adjust this by adding "-march=native" to your compile options.

**Comment by cookpa** [comment]
Do we think any of the recent CMake changes have fixed this?

**Comment by gdevenyi** [comment]
No, the issue is that a commit from @hjmjohnson set the lowest level of instruction sets to "new" CPUs:
https://github.com/ANTsX/ANTs/blob/bcab74485ee07ca940eca59e6c22c4d7d991af44/CMake/ITKSetStandardCompilerFlags.cmake#L241-L247

The solution is the user can override these settings.

**Comment by cookpa** [comment]
I found a machine that produces this error, so I looked into it a bit. Adding

```
  -DSuperBuild_ANTS_C_OPTIMIZATION_FLAGS="-mtune=native -march=native" \
  -DSuperBuild_ANTS_CXX_OPTIMIZATION_FLAGS="-mtune=native -march=native" 
```

to the cmake call worked (using GCC 4.9.1 on Linux). I guess for portable builds one would want to replace "native" with "x86-64".

It looks like "corei7" has been removed from the list of GCC supported architectures:

https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html

So I'm not sure what a modern GCC will do with "-march=corei7".
